integration/spark/app/src/test/java/io/openlineage/spark/agent/lifecycle/MatchesMapRecursively.java

```diff
@@ -0,0 +1,117 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.spark.agent.lifecycle;
+
+import io.openlineage.spark.agent.Versions;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import lombok.extern.slf4j.Slf4j;
+import org.assertj.core.api.AbstractObjectAssert;
+import org.assertj.core.api.Condition;
+
+/**
+ * Custom Condition writen for recursive comparison of Map and List with ability to ignore specified
+ * Map keys AssertJ Built-in recursive comparison is not working with Map type, it can ignore only
+ * object properties Example usage: assertThat(actualMap).satisfies(new
+ * MatchesMapRecursively(expectedMap, new HashSet<>(Arrays.asList("runId"))));
+ *
+ * @see AbstractObjectAssert#usingRecursiveComparison()
+ */
+@Slf4j
+public class MatchesMapRecursively extends Condition<Map<String, Object>> {
```

Comment:
Would be nice to have shared test code for this case, rather than duplicate code. Let's do it in other PR though.